### ISSUE TITLE
Remove microprofile-telemetry proposal

### ISSUE THREAD

**Issue by brunobat** [eclipse/microprofile-sandbox#130]
The `microprofile-telemetry` proposal was already accepted and[ the related module](https://github.com/eclipse/microprofile-sandbox/tree/main/proposals/microprofile-telemetry) should be removed.
See: https://github.com/eclipse/microprofile-telemetry

**Comment by brunobat** [comment]
CC @Emily-Jiang

**Comment by Emily-Jiang** [comment]
@brunobat yes, we should update this repo by removing the accepted proposals.
